derive/src/lib.rs

```diff
@@ -103,7 +104,112 @@ pub fn plugin_fn(
     }
 }
 
-/// `host_fn` is used to define a host function that will be callable from within a plugin
+/// `export_fn` is used to define a function that will be exported by a plugin, but is not directly
+/// callable by an Extism runtime. These functions can be used for runtime linking and mocking host
+/// functions for tests. If direct access to Wasm native parameters is needed, then a bare
+/// `extern "C" fn` should be used instead.
+///
+/// All arguments should implement `extism_pdk::ToBytes` and the return value should implement
+/// `extism_pdk::FromBytes`
+#[proc_macro_attribute]
+pub fn export_fn(
```

Comment:
I'm a little unsure of name. The docs make it clear. But it's hard to tell which to use just from the name. I can't think of a better name though.

Reply:
Yea, agreed that although the docs make it clear, the name is a little ambiguous. 

What about something like `#[linkable_fn]`? That is, a function which can be linked by Extism. I had also thought about `#[shared_fn]`, since like a shared library, you don't execute it via a main entrypoint, but through some other execution path some code finds a function and calls it?

I think my main concern is that an Extism user, with even rudimentary knowledge of Wasm is going to see `#[export_fn]` and mistake it for `#[plugin_fn]`, since the name correctly suggests the function will be exported.